portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,30 +68,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
      * Note: commeentMap data structure: 
      * {userEmail: [[userComment1, userCommentImage1],[userComment2, userCommentImage2]...]}
      */
-    HashMap<String, ArrayList<ArrayList<String>>> userComment = new HashMap<String, ArrayList<ArrayList<String>>>();
+    HashMap<String, ArrayList<ArrayList<String>>> userComment = new HashMap();
```

Comment:
I don't really like the idea of ArrayList<ArrayList>> when you know the size of the inside ArrayList is fixed. I think a class called UserComment to contain both fields is preferred here. If it is hard to parse it to json, maybe we can try java tuple or some equivalent data structures?

Reply:
agreed. A class to encapsulate the comment and picture would look better.